IPython/terminal/console/completer.py

```diff
@@ -36,8 +37,17 @@ def __init__(self, shell, client, config=None):
             self.readline.set_completer_delims('\r\n')
     
     def complete_request(self, text):
-        line = readline.get_line_buffer()
-        cursor_pos = readline.get_endidx()
+        line = str_to_unicode(readline.get_line_buffer())
+        byte_cursor_pos = readline.get_endidx()
+        
+        # get_endidx is a byte offset
+        # account for multi-byte characters to get correct cursor_pos
+        cursor_pos = byte_cursor_pos
+        i = 0
+        while i < cursor_pos:
+            bytelen = len(cast_bytes(line[i]))
+            cursor_pos -= (bytelen-1)
+            i += 1
```

Comment:
that makes sense. If encoding and/or decoding fails, you might get off by one, but I'm not too worried about that.